clash-lib/src/Clash/Util.hs

```diff
@@ -295,24 +254,6 @@ indexNote
   -- ^ Error or element /n/
 indexNote note = \xs i -> fromMaybe (error note) (indexMaybe xs i)
 
--- | Safe version of 'head'
-headMaybe :: [a] -> Maybe a
-headMaybe (a:_) = Just a
-headMaybe _ = Nothing
-
--- | Safe version of 'tail'
-tailMaybe :: [a] -> Maybe [a]
-tailMaybe (_:as) = Just as
-tailMaybe _ = Nothing
```

Comment:
IMO keep these around, they're useful no?

Reply:
They were only used in one place and both were used (equivalent to uncons).
I also don't think `fmap fst . uncons` etc. is any more work to write :man_shrugging:

Reply:
I wasn't aware of `uncons` (and it certainly a  better fit for where it's used). Having said that, `headMaybe` is definitely easier to write, but more importantly read, than `fmap fst . uncons`.

Reply:
True, but there's also the extra where do I import that from / where is that imported from which is avoided with `uncons`. Given it was only used in one place I don't think this is much of a loss anyway